crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2957,6 +2962,54 @@ impl Room {
             .await?;
         Ok(())
     }
+
+    /// Remove verification requirements for the given users and
+    /// resend messages that failed to send because their identities were no
+    /// longer verified (in response to
+    /// `SessionRecipientCollectionError::VerifiedUserChangedIdentity`)
+    #[cfg(feature = "e2e-encryption")]
+    pub async fn withdraw_verification_and_resend(
+        &self,
+        user_ids: Vec<OwnedUserId>,
+        transaction_id: &TransactionId,
+    ) -> Result<()> {
+        let encryption = self.client().encryption();
+
+        for user_id in user_ids {
+            if let Some(user_identity) = encryption.get_user_identity(&user_id).await? {
+                user_identity.withdraw_verification().await?;
+            }
+        }
+
+        self.send_queue().unwedge(transaction_id).await?;
+
+        Ok(())
+    }
+
+    /// Set the local trust for the given devices to `LocalTrust::Verified`
+    /// and resend messages that failed to send because said devices are
+    /// unverified (in response to
+    /// `SessionRecipientCollectionError::VerifiedUserHasUnsignedDevice`).
+    #[cfg(feature = "e2e-encryption")]
+    pub async fn trust_devices_and_resend(
```

Comment:
I guess this should probably then become something like `ignore_device_trust_and_resend`